### PR TITLE
WIP: Bleatbot

### PR DESCRIPTION
**Pull request by NotNite**
From #37, I present to you: Bleatbot!

Our very own beepy-boopy GitHub bot. See the (very lazily written) DIP for info.

I expect to revise the DIP in a bit, feel free to comment if I was too lazy at points. Also I stole a bunch of text from DIP29.

### CODE REVIEW

File: text/41-bleatbot.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+In the comments of a issue/pull request on a goatcorp GitHub repository, members of the goatcorp GitHub group can execute any of the following commands:
+
+- `@bleatbot faq <entry>` - Post an FAQ entry.
+- `@bleatbot autoClose <when>` - Automatically close an issue after the specified time (e.g `24h`).
+- `@bleatbot autoMerge <when>` - Automatically merge a pull request after the specified time (e.g. `24h`).
+- `@bleatbot cancelAuto` - Cancel any automatic tasks.
```
Comment (philpax) [inline text/41-bleatbot.md:31]: For these three, I'd suggest `autoclose`, `automerge` and `cancel`. Probably worth having a look at https://bors.tech/documentation/.

**Comment by philpax** [comment]
Sorry, should've gotten back to you earlier. This seems pretty reasonable to me - might be nice to get opinions from other members of @goatcorp/community-moderation to see if they have any needs they think should be addressed.

File: text/41-bleatbot.md (hunk 1 above)
Comment (NotNite) [reply text/41-bleatbot.md:31]: Will change when I'm free.

**Comment by NotNite** [comment]
I ask of someone to invite the bot to the GitHub organization so I can check who can use it based on if they're in the organization.

**Comment by philpax** [comment]
I've taken the liberty of inviting bleatbot. pls don't aboos api access

This also looks good for inspiration on automating the DIPs if we want to: https://github.com/rust-lang/rfcbot-rs

**Comment by NotNite** [comment]
Maybe I'll clean up my profanity in the Git history and we can make a `goatcorp/bleatbot` repo so other people can help contribute. I'm a bit bad at writing good code... lol

**Comment by NotNite** [comment]
> I ask of someone to invite the bot to the GitHub organization so I can check who can use it based on if they're in the organization.

Implemented this. My Git history is a bit messy so I'm definitely going to wipe it when I publish it, but it should be not-awful-enough to go on `goatcorp/bleatbot` if we're okay with making it.

**Comment by philpax** [comment]
@bleatbot exists but doesn't implement this DIP, afaik. Do we close this?

**Comment by NotNite** [comment]
No. It implements this DIP. I am waiting on goat's approval to move to its own repository under goatcorp/.

**Comment by philpax** [comment]
What's doing the `bleatbot, approve` stuff then?

**Comment by NotNite** [comment]
I believe a serverless function? It should ideally be moved to bleatbot's TS codebase in the future.

**Comment by philpax** [comment]
@goaaats can you provide some clarity around what @bleatbot is doing/running and where the code for that lives? Once that's clarified we'll unify that with the bleatbot codebase, update this DIP, merge it in, and bring bleatbot into the corp

**Comment by goaaats** [comment]
The only thing I run atm re: bleatbot is a CF worker that dispatches builds for plogon, here's the code(has CD set up): https://github.com/goatcorp/worker-plugindispatch/actions

**Comment by philpax** [comment]
@NotNite do you think it'd be possible to integrate that dispatcher into your code? Does it need to be integrated?

**Comment by NotNite** [comment]
I'm not sure if it needs to be integrated, but it would probably be easy to pull off. I think the real concern is finding people to help develop Bleatbot as I'm starting school soon.

**Comment by philpax** [comment]
That's fine, we can just get it up and running and build up new features as required. If we start on #46, deploying bleatbot to somewhere on goatcorp infra should be pretty easy. Given there's no general objection to the concept or implementation, I'm going to merge this and we can work out the specifics of implementation as we go.

**Comment by Bluefissure** [comment]
> The only thing I run atm re: bleatbot is a CF worker that dispatches builds for plogon, here's the code(has CD set up): https://github.com/goatcorp/worker-plugindispatch/actions

Hi @goaaats and @NotNite, I worked on some of the rebuild functions into bleatbot to trigger the workflow. https://github.com/ottercorp/bleatbot/commit/66469e949fd2997e3be1b647169863050a74dcbb

Since I don't have access to the `worker-plugindispatch` repo I can only guess the behaviour of the command. No cooldown is introduced for now.

If it looks okay I can make a PR to bleatbot, but I think it's better to compare it with `worker-plugindispatch` first.

**Pull request merged**